.github/workflows/preview-env-deploy.yml

```diff
@@ -2,6 +2,9 @@ name: preview-env-deploy
 on:
   pull_request:
     types: [labeled, synchronize]
+  push:
+    branches:
+      - master
```

Comment:
I'd consider moving this logic to a dedicated workflow with a clearly distinctive name like `preview-env-build-cache` to avoid mixing concerns (this workflow doesn't deploy `main`) as that could lead to mistakes later:

All other `preview-env-deploy` workflows that the Infra team maintains are not ran on `main` so the logic is surprising for the reader and someone editing this file might e.g. forget the `if: github.ref != 'refs/heads/main'` condition on a newly added step.

Reply:
This is actually a very good point.